app/Http/Controllers/UserController.php

```diff
@@ -15,23 +17,54 @@ public function get(User $user): User
     public function store(Request $request)
     {
         $request->validate([
-            'id' => 'nullable',
-            'username' => 'required|min:3|max:255',
-            'location' => 'nullable|regex:/^([-+]?)([\d]{1,2})(((\.)(\d+)(,)))(\s*)(([-+]?)([\d]{1,3})((\.)(\d+))?)$/i'
+            'username'   => 'required|min:3|max:255',
+            'location'   => 'nullable|regex:/^([-+]?)([\d]{1,2})(((\.)(\d+)(,)))(\s*)(([-+]?)([\d]{1,3})((\.)(\d+))?)$/i',
+            'invite_key' => 'required|exists:invite_keys,value',
         ]);
-        $user = new User();
-        $user->id = $request->get('id');
-        $user->username = $request->get('username');
-        $user->location = $request->get('location');
+
+        $inviteKeyId = $request->get('invite_key');
+
+        // Check if user already exists with inviteKey
+        if (User::query()->where('invite_key', $inviteKeyId)->count() > 0) {
+            return null;
+        }
+        $role = $request->get('role');
+
+        $user = User::create([
+            'username'   => $request->get('username'),
+            'location'   => $request->get('location'),
+            'invite_key' => $inviteKeyId,
+            'role'       => isset($role) ? $role : Roles::Thief,
+        ]);
+
         $user->save();
+
+        return $user;
     }
 
     public function update(Request $request, User $user)
     {
         $request->validate([
-            'location' => 'nullable|regex:/^([-+]?)([\d]{1,2})(((\.)(\d+)(,)))(\s*)(([-+]?)([\d]{1,3})((\.)(\d+))?)$/i'
+            'location' => 'nullable|regex:/^([-+]?)([\d]{1,2})(((\.)(\d+)(,)))(\s*)(([-+]?)([\d]{1,3})((\.)(\d+))?)$/i',
         ]);
         $user->location = $request->get('location');
         $user->save();
     }
+
+    // Return InviteKey based on value
+    // Key could be null, if InviteKey doesn't exists anymore
+    public function getInviteKey($inviteKeyId)
+    {
+        $inviteKey = InviteKey::query()->where('value', $inviteKeyId)->first();
+
+        if (! isset($inviteKey)) {
+            return null;
+        }
+
+        // TODO: Player could have an InviteCode with the same value, but then InviteCode can't have same TeamId
+        $totalInUse = User::query()->where('invite_key', $inviteKey->value)->count();
+
+        // Check if InviteKey not yet in use
+        return $totalInUse == 0 ? $inviteKey : null;
+    }
```

Comment:
Klopt het dat deze functie in de browser een lege html pagina returnt? Als ik hier een dd($totalInUse == 0) plaats krijg ik "false" terug, maar die zie je niet in de browser.

Reply:
Dat klopt dan. Als de `inviteKey` al in gebruik is, dan is `$totalInUse == 0` false en dan stuurt de website `null` terug. Als speler kan er maar één `InviteKey` worden gebruikt.

Reply:
Ik had specifiek gecontroleerd en hij zou in mijn geval geen null hebben moeten returnen. Hij returnt ook wel een value, maar gewoon niet als tekst op de pagina. Lijkt me dat de endpoint op zijn minst iets van JSON of een String zou returnen.

Reply:
Ja dat kan, alleen weet ik niet zo 123 wat ik dan moet terugsturen. Er wordt namelijk een request gemaakt om een `InviteKey` op te halen, als dat niet mag of die bestaat niet moet die dan misschien een foutmelding geven? Als die dan niet bestaat een 404 en anders als die al in gebruik is, dan geen toegang verlenen, dus een 403

Reply:
Je kunt een JSON object terugsturen met 
a.) de key
b.) error: key can't be used

ofzo